### ISSUE TITLE
Warn about unknown type in return value

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#468]
**Is your feature request related to a problem? Please describe.**
When returning an unknown type value from an expression, the value is implicitly cast to the return value type from the function definition.  This can happen when, for example, the stubs have not been fully fleshed out (as is the case with SQLAlchemy).  To illustrate:

```py
# pyright: strict, reportMissingImports=false

from foo import bar


def implicit_cast_1() -> int:
    return bar    # Passes


def implicit_cast_2() -> int:
    wat = bar     # Caught in strict mode or with 'reportUnknownVariableType' enabled
    return wat
```

This produces:

```json
{
	"resource": "test.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "Inferred type of 'wat' is unknown",
	"source": "pyright (reportUnknownVariableType)",
	"startLineNumber": 11,
	"startColumn": 5,
	"endLineNumber": 11,
	"endColumn": 8
}
```

It would be beneficial to warn users when the inferred return value type does not match the return value type in the function annotation independently of how it is derived.

**Describe the solution you'd like**
An option to report that the inferred return value type is unknown.

**Comment by erictraut** [comment]
Thanks for the suggestion. This will be implemented in the next version of Pyright.

**Comment by erictraut** [comment]
This is implemented in 1.1.18, which I just published. It's controlled by the reportUnknownVariableType setting. I decided not to provide another switch specifically for return expressions.
